src/document_graph/creators.cpp

```diff
@@ -73,6 +75,35 @@ namespace hyphaspace {
         return create_document (creator, new_content(content_label, content_value));
     }
 
+    document_graph::document document_graph::update_document(const name &updater, const checksum256 &doc_hash, vector<content_group> content_groups)
```

Comment:
This updates the document and its hash, but has the potential to orphan edges that will no longer point to this document. Should this also update the edges table with all references to this document in from and to?

Reply:
Yes, I thought of that also, but wasn't sure about placing all that secondary login within the function. 

I could add that logic and maybe change the name of the function to update_document_and_edges, so just to make it clear that it also will update dependent edges. What do you think?